tests/subsys/nrf_compress/decompression/mcuboot_update/src/main.c

```diff
@@ -13,16 +13,23 @@
 #include <zephyr/sys/byteorder.h>
 
 #include <bootutil/image.h>
+#include <zephyr/devicetree.h>
 
 LOG_MODULE_REGISTER(mcuboot_update, LOG_LEVEL_DBG);
 
-#define UPDATE_SLOT_NUMBER CONFIG_UPDATE_SLOT_NUMBER
-
-#if CONFIG_UPDATE_SLOT_NUMBER == 0
+#if defined(CONFIG_PARTITION_MANAGER_ENABLED)
+#if CONFIG_UPDATE_IMAGE_NUMBER == 0
 #define UPDATE_SLOT_ID PM_MCUBOOT_SECONDARY_ID
 #else
-#define UPDATE_SLOT_ID PM_MCUBOOT_SECONDARY_ ## CONFIG_UPDATE_SLOT_NUMBER ## _ID
-#endif
+#define UPDATE_SLOT_ID PM_MCUBOOT_SECONDARY_ ## CONFIG_UPDATE_IMAGE_NUMBER ## _ID
+#endif /* #if CONFIG_UPDATE_IMAGE_NUMBER == 0 */
+#else
+#if CONFIG_UPDATE_IMAGE_NUMBER == 0
+#define UPDATE_SLOT_ID DT_FIXED_PARTITION_ID(DT_NODELABEL(slot1_partition))
+#else
+#define UPDATE_SLOT_ID DT_FIXED_PARTITION_ID(DT_NODELABEL(slot3_partition))
+#endif /* #if CONFIG_UPDATE_IMAGE_NUMBER == 0 */
+#endif /* #if defined(CONFIG_PARTITION_MANAGER_ENABLED) */
```

Comment:
There is no need to add `#if` in the comment at the end of the block.
Simply - keep the condition, that's enough to help the Reader :slightly_smiling_face: